event_test.go

```diff
@@ -253,6 +257,37 @@ func TestEventfLogMetadataProvider(t *testing.T) {
 	assert.EqualValues(t, expected, e.Metadata)
 }
 
+func TestSerializeDeserialize(t *testing.T) {
+	event := Event{
+		Context:         context.Background(),
+		Id:              "test",
+		Timestamp:       time.Now(),
+		Severity:        ErrorSeverity,
+		Message:         "foo",
+		OriginalMessage: "foo",
+		Metadata: map[string]interface{}{
+			"string": "value",
+			"number": float64(42),
+		},
+		Labels: map[string]string{
+			"label": "foo",
+		},
+		Error: errors.New("an error"),
+	}
+	out, err := json.Marshal(&event)
+	assert.NoError(t, err)
+
+	var undo Event
+	err = json.Unmarshal(out, &undo)
+	assert.NoError(t, err)
+
+	assert.Equal(t, event.Id, undo.Id)
+	assert.Equal(t, event.Severity, undo.Severity)
+	assert.Equal(t, event.Message, undo.Message)
+	assert.Equal(t, event.Metadata, undo.Metadata)
+	assert.Equal(t, event.Labels, undo.Labels)
```

Comment:
Am I right in thinking we're expecting `event.Error` to be empty when it's unmarshaled?

Reply:
Yeah, the intention is just that we don't error here. Anybody who wants to marshal/unmarshal the error field needs to do something special with their error type.

Reply:
This is a nit pick, but is it worth adding an assertion for this field and including that as a comment?

Reply:
It just confused me that we weren't asserting at all.